tests/kafkatest/services/console_consumer.py

```diff
@@ -226,10 +230,13 @@ def _worker(self, idx, node):
 
             for line in itertools.chain([first_line], consumer_output):
                 msg = line.strip()
-                if self.message_validator is not None:
-                    msg = self.message_validator(msg)
-                if msg is not None:
-                    self.messages_consumed[idx].append(msg)
+                if msg == "shutdown_complete":
+                    self.nodes_clean_shutdown.append(idx)
+                else:
+                    if self.message_validator is not None:
+                        msg = self.message_validator(msg)
+                    if msg is not None:
+                        self.messages_consumed[idx].append(msg)
```

Comment:
Ah, I was actually thinking that any message (in particular a consumed message?) after a shutdown complete would be a thing to raise an error over.

Really just a sanity check, might not actually be of use.